server/game/cards/05_LOF/leaders/ThirdSisterSeethingWithAmbition.ts

```diff
@@ -0,0 +1,59 @@
+import AbilityHelper from '../../../AbilityHelper';
+import * as AbilityLimit from '../../../core/ability/AbilityLimit';
+import type { TriggeredAbilityContext } from '../../../core/ability/TriggeredAbilityContext';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { CardType, Duration, EventName, KeywordName, RelativePlayer, WildcardCardType, ZoneName } from '../../../core/Constants';
+
+export default class ThirdSisterSeethingWithAmbition extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: 'lof-preview-third-sister-seething-with-ambition',
+            internalName: 'third-sister#seething-with-ambition',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'Play a unit from your hand. It gains Hidden for this phase',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            targetResolver: {
+                cardCondition: (card) => card.isUnit(),
+                cardTypeFilter: CardType.BasicUnit,
+                controller: RelativePlayer.Self,
+                zoneFilter: ZoneName.Hand,
+                immediateEffect: AbilityHelper.immediateEffects.simultaneous([
+                    AbilityHelper.immediateEffects.playCardFromHand({
+                        playAsType: WildcardCardType.Unit,
+                    }),
+                    AbilityHelper.immediateEffects.forThisPhaseCardEffect({
+                        effect: AbilityHelper.ongoingEffects.gainKeyword(KeywordName.Hidden)
+                    }),
+                ])
+            }
+        });
+    }
+
+    protected override setupLeaderUnitSideAbilities() {
+        this.addOnAttackAbility({
+            title: 'The next unit you play this phase gains Hidden',
+            immediateEffect: AbilityHelper.immediateEffects.delayedPlayerEffect({
+                title: 'The next unit you play this phase gains Hidden',
+                when: {
+                    onCardPlayed: this.isUnitPlayedEvent,
+                },
+                duration: Duration.UntilEndOfPhase,
+                limit: AbilityLimit.perGame(1),
+                immediateEffect: AbilityHelper.immediateEffects.forThisPhaseCardEffect((context) => ({
```

Comment:
I realized this is wrong, the on attack ability gives Hidden beyond the current phase which can be important for cards that check the existence of a keyword. Will fix it later!